subprojects/platform-jvm/src/main/java/org/gradle/api/java/archives/internal/DefaultManifest.java

```diff
@@ -224,6 +226,110 @@ static void writeTo(org.gradle.api.java.archives.Manifest manifest, OutputStream
         }
     }
 
+    // Pulled from java.util.jar.Manifest
+    @SuppressWarnings("deprecation")
+    private static void write(Manifest manifest, OutputStream out) throws IOException {
+        DataOutputStream dos = new DataOutputStream(out);
+        // Write out the main attributes for the manifest
+        writeMain(manifest.getMainAttributes(), dos);
+        // Now write out the per-entry attributes
+        for (Map.Entry<String, java.util.jar.Attributes> e : manifest.getEntries().entrySet()) {
+            StringBuffer buffer = new StringBuffer("Name: ");
+            String value = e.getKey();
+            if (value != null) {
+                byte[] vb = value.getBytes(StandardCharsets.UTF_8);
+                value = new String(vb, 0, 0, vb.length);
+            }
+            buffer.append(value);
+            make72Safe(buffer);
+            buffer.append("\r\n");
+            dos.writeBytes(buffer.toString());
+            write(e.getValue(), dos);
+        }
+        dos.flush();
+    }
+
+    // Pulled from java.util.jar.Attributes
```

Comment:
`java.util.jar.Attributes` is GPL with classpath exception.
Is that compatible with Gradle's Apache 2.0 license? I don't think so.

Reply:
It's a valid point. I will close this PR given the license are indeed uncompatible. However, the extent of the fix should be along those line. We could path the JDK implementation if the `make72Safe` method wasn't static on `java.util.jar.Manifest`.